lib/BC/SleighLifter.cpp

```diff
@@ -700,24 +736,65 @@ class SleighLifter::PcodeToLLVMEmitIntoBlock : public PcodeEmit {
   static std::unordered_set<OpCode> INTEGER_COMP_OPS;
 
 
+  struct VisitExit {
+    PcodeToLLVMEmitIntoBlock &th;
+    llvm::BasicBlock *operator()(const sleigh::InstrExit &exit) {
+      return th.exit_block;
+    };
+
+    llvm::BasicBlock *operator()(const sleigh::IntrainstructionIndex &exit) {
+      return th.GetOrCreateBlock(exit.target_block_ind);
+    };
+  };
+  struct VisitBlockExitTrue {
+    PcodeToLLVMEmitIntoBlock &th;
+    llvm::BasicBlock *operator()(const sleigh::Exit &exit) {
+      return std::visit(VisitExit{th}, exit);
+    }
+    llvm::BasicBlock *operator()(const sleigh::ConditionalExit &exit) {
+      return std::visit(VisitExit{th}, exit.true_branch);
+    }
+  };
+
+  struct VisitBlockExitFalse {
+    PcodeToLLVMEmitIntoBlock &th;
+    llvm::BasicBlock *operator()(const sleigh::Exit &exit) {
+      return std::visit(VisitExit{th}, exit);
+    }
+    llvm::BasicBlock *operator()(const sleigh::ConditionalExit &exit) {
+      return std::visit(VisitExit{th}, exit.false_branch);
+    }
+  };
```

Comment:
Copy & pasted 2 times (almost 3). Extract all shared parts ( attribute + `Exit` visit) into base class from which then inherit. Figure out, whether the other two should also inherit from the first one to inherit those visits.

Reply:
I don't really understand the machinery of `std::visit` but it doesn't seem to accept the `operator` overloads if they're in a base class. As soon as I move the shared `Exit` visit in a base class, the compiler complains about the visitor not being exhaustive.

Reply:
[892cada](https://github.com/lifting-bits/remill/pull/645/commits/892cadada6d35d71e58253fed4f02cf28fc497cf)